src/drivers/rover_interface/RoverInterface.cpp

```diff
@@ -0,0 +1,378 @@
+#include "RoverInterface.hpp"
+#include <px4_platform_common/log.h>
+
+RoverInterface *RoverInterface::_instance;
+
+RoverInterface::RoverInterface(uint8_t rover_type, const char *can_iface)
+	: ModuleParams(nullptr),
+	ScheduledWorkItem(MODULE_NAME, px4::wq_configurations::rover_interface),
+	_rover_type(rover_type)
+{
+	pthread_mutex_init(&_node_mutex, nullptr);
+	_can_iface = strdup(can_iface);
```

Comment:
try to drop the strdup; you can either assume that the can_iface is constant (like it is now in this driver), or if you actually need to store it, allocate a char buffer in the class and copy it in there.

The reasoning is that by spec strdup needs to be freed by "free"; but in nuttx protected build/kernel build this driver being inside the kernel, it is actually allocated from the kernel heap and from there it can't be freed.. (would need kmm_free or such). If you actually need to even store it?

Reply:
yes, converted to const char buffer in class